source/GameInterface/Services/CultureObjects/Patches/CultureObjectLifetimePatches.cs

```diff
@@ -0,0 +1,43 @@
+﻿using Common.Logging;
+using Common.Messaging;
+using GameInterface.Policies;
+using GameInterface.Services.BasicCultureObjects.Messages;
+using GameInterface.Services.ObjectManager;
+using HarmonyLib;
+using Serilog;
+using System;
+using TaleWorlds.CampaignSystem;
+
+namespace GameInterface.Services.BasicCultureObjects.Patches
+{
+    [HarmonyPatch]
+    internal class CultureObjectLifetimePatches
+    {
+        private static readonly ILogger Logger = LogManager.GetLogger<CultureObjectLifetimePatches>();
+
+        [HarmonyPatch(typeof(CultureObject), MethodType.Constructor)]
```

Comment:
Add destruction postfix patches for the following
- CreatePreCraftedWeapon
- InitializePreCraftedWeaponOnLoad
- CreateRandomCraftedItem

Add a destruction prefix for CraftingState.CraftingLogic setter, and delete the current CraftingLogic value (if it is not null)